+p_xml: 'version="1.0" encoding="UTF-8" standalone="yes" '
+directive: DOCTYPE bugzilla SYSTEM "http://bugzilla.fieldtriptoolbox.org/page.cgi?id=bugzilla.dtd"
bugzilla:
  +@version: 4.4.1
  +@urlbase: http://bugzilla.fieldtriptoolbox.org/
  +@maintainer: r.oostenveld@donders.ru.nl
  bug:
    bug_id: "1433"
    creation_ts: 2012-04-18 15:15:00 +0200
    short_desc: test_ft_datatype_sens fails
    delta_ts: 2012-12-31 11:46:25 +0100
    reporter_accessible: "1"
    cclist_accessible: "1"
    classification_id: "1"
    classification: Unclassified
    product: FieldTrip
    component: core
    version: unspecified
    rep_platform: PC
    op_sys: Windows
    bug_status: CLOSED
    resolution: FIXED
    bug_file_loc:
    status_whiteboard:
    keywords:
    priority: P3
    bug_severity: normal
    target_milestone: '---'
    everconfirmed: "1"
    reporter:
      +content: johanna.zumer
      +@name: Johanna
    assigned_to:
      +content: r.oostenveld
      +@name: Robert Oostenveld
    cc: r.oostenveld
    comment_sort_order: oldest_to_newest
    long_desc:
      - +@isprivate: "0"
        commentid: "5926"
        comment_count: "0"
        who:
          +content: johanna.zumer
          +@name: Johanna
        bug_when: 2012-04-18 15:15:06 +0200
        thetext: |-
          It first fails for i=3 fullfile('neuromag122', 'jg_single_01raw.fif').  It seems from SVN revision 4670, there was a check in test_ft_datatype_sens that if 'unit' not exist on sens2, then add the 'unit' of sens1.  This works for all data cases except i=3, because this data is empty (all the fields exist, but are size 0x3). ft_convert_units requires that fields not be empty.

          Perhaps a bigger question is, why do both sens1 and sens2 for i=3 have empty fields?

          It secondly fails for i=16   fullfile('yokogawa440', 'S1_MEG_Epoch.raw').  sens1 (on disk) shows normal values, but sens2 some fields are zero (e.g. chanori is the correct size but all zeros).  I traced it to line 99 of ft_datatype_sens where sens.ori makes sense but then the output of chanori is all zeros.

          I think Robert may know the answer to i=3, and JM for i=16 ...?
      - +@isprivate: "0"
        commentid: "5928"
        comment_count: "1"
        who:
          +content: johanna.zumer
          +@name: Johanna
        bug_when: 2012-04-18 15:17:17 +0200
        thetext: '*** Bug 1417 has been marked as a duplicate of this bug. ***'
      - +@isprivate: "0"
        commentid: "8827"
        comment_count: "2"
        who:
          +content: r.oostenveld
          +@name: Robert Oostenveld
        bug_when: 2012-12-31 11:25:32 +0100
        thetext: "The test does not show up as failed on the dashboard, since it has a try-catch and then shows a warning instead of an error.\n\nI reran the test and it fails for\n\nK>> i\ni =\n    16\nK>> filename(i)\nans = \n    'yokogawa440/S1_MEG_Epoch.raw'\n\nThe failure is due to chanpos and ori being all zeros for sens2"
      - +@isprivate: "0"
        commentid: "8828"
        comment_count: "3"
        who:
          +content: r.oostenveld
          +@name: Robert Oostenveld
        bug_when: 2012-12-31 11:39:30 +0100
        thetext: |-
          (In reply to comment #2)

          correction: only the chanori is all zeros in sens2, but the chanpos is still different between sens1 and sens2.
      - +@isprivate: "0"
        commentid: "8829"
        comment_count: "4"
        who:
          +content: r.oostenveld
          +@name: Robert Oostenveld
        bug_when: 2012-12-31 11:44:15 +0100
        thetext: "There only exists one 440-channel yokogawa system in the world and it has a complicated mixture of channel types. I would rather not spend time on debugging this system at the moment.\n\n I have removed the warning and made it into an error, that makes the script show up in the dashboard. I have also updated the test so that the known discrepancy for yokogawa440 does not trigger an error.\n\nroboos@mentat001> svn commit test/test_ft_datatype_sens.m \nSending        test/test_ft_datatype_sens.m\nTransmitting file data .\nCommitted revision 7273."
      - +@isprivate: "0"
        commentid: "8848"
        comment_count: "5"
        who:
          +content: r.oostenveld
          +@name: Robert Oostenveld
        bug_when: 2012-12-31 11:46:25 +0100
        thetext: closed several bugs that have been resolved for some time. Feel free to reopen the bug if you disagree.
